packages/fields/src/actions/pattern/delete-pattern.tsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * WordPress dependencies
+ */
+import { trash } from '@wordpress/icons';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { privateApis as patternsPrivateApis } from '@wordpress/patterns';
+import type { Action } from '@wordpress/dataviews';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from '../utils';
+import type { Pattern } from '../../types';
+import { decodeEntities } from '@wordpress/html-entities';
+import { unlock } from '../../lock-unlock';
+import type { NoticeSettings } from '../../mutation';
+import { deletePostWithNotices } from '../../mutation';
+
+const { PATTERN_TYPES } = unlock( patternsPrivateApis );
+
+const deletePatternAction: Action< Pattern > = {
+	id: 'delete-pattern',
+	label: __( 'Delete' ),
+	isPrimary: true,
+	icon: trash,
+	isEligible( post ) {
+		return post.type === PATTERN_TYPES.user;
+	},
+	supportsBulk: true,
+	hideModalHeader: true,
+	RenderModal: ( { items, closeModal, onActionPerformed } ) => {
```

Comment:
Could the `RenderModal` for this one and `delete-template` be shared?
I realize currently this doesn't have access to the above action, but maybe we can move this into a thin wrapper?
A future idea is to pass the action into the `RenderModal` https://github.com/WordPress/gutenberg/blob/trunk/packages/dataviews/src/components/dataviews-item-actions/index.tsx#L115

Reply:
>Could the RenderModal for this one and delete-template be shared?

Good point! I would keep them separate and if we will see an overlap with other entities try to create a shared modal.